File: Jan23/Ex0.cpp
#include <iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;
    long numer, denom, decimals, tempRes, res;
    for (int i = 0; i < t; i++)
    {
        cin >> numer >> denom >> decimals;
        res = 1;
        numer %= denom;
        for (int j = 0; j < decimals; j++)
        {
            numer *= 10;
            tempRes = numer / denom;
            res *= (tempRes == 0) ? 1 : tempRes;
            res %= 188888881;
            numer %= denom;
        }
        cout << res << endl;
    }
    return 0;
}

File: Jan23/Ex0.java
import java.util.Scanner;

public class Ex0 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        long numer, denom, decimals, tempRes, res;

        for (int i = 0; i < t; i++) {
            numer = sc.nextInt();
            denom = sc.nextInt();
            decimals = sc.nextInt();
            res = 1;
            numer %= denom;
            for (int j = 0; j < decimals; j++) {
                numer *= 10;
                tempRes = numer / denom;
                res *= (tempRes == 0) ? 1 : tempRes;
                numer %= denom;
            }
            System.out.println(res);
        }
        sc.close();
    }
}